tough/src/schema/mod.rs

```diff
@@ -34,10 +36,25 @@ forward_display_to_serde!(RoleType);
 forward_from_str_to_serde!(RoleType);
 
 /// Common trait implemented by all roles.
-pub trait Role {
+pub trait Role: Serialize {
     const TYPE: RoleType;
 
     fn expires(&self) -> DateTime<Utc>;
+
+    fn canonical_form(&self) -> Result<Vec<u8>> {
+        canonical_form_impl(self)
```

Comment:
And now that we're only calling it from one place, we can collapse the convenience function into here :)

Reply:
😳  Oops... yeah that makes sense.